week1/reposts/boccato/Main.hs

```diff
@@ -0,0 +1,38 @@
+module Main where
+
+import Control.Monad
+import Data.List
+import System.Random
+import Text.Printf
+
+nWinners = 7
+nSimulations = 10000
+nCurrentReposts = 1000000
+
+takeDistinct :: Eq a => Int -> [a] -> [a]
+takeDistinct n xs = sample n xs []
+  where
+    sample _ []     res  = res
+    sample n (x:xs) res
+      | n == 0       = res
+      | x `elem` res = sample n xs res
+      | otherwise    = sample (n-1) xs (x:res)
+
+-- nWinners are taken from a pool of n+nCurrentReposts tickets
+simulate :: Int -> IO Bool
```

Comment:
I believe the function could be "purer" (and easier to test?) if the generator was one of the arguments to the function, instead of getting the default global one inside it.